server/handlers/componetgenratorhelper.go

```diff
@@ -0,0 +1,647 @@
+
+	if !checkOCI {
+		if err := utils.ExtractFile(filePath, tempDir); err != nil {
+			h.sendErrorEvent(uuid.Nil, provider, "Error creating temp directory", err)
+			return err
+		}
+	}
+	var wg sync.WaitGroup
+	var mu sync.Mutex
+	if err := filepath.Walk(tempDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return meshkitutils.ErrFileWalkDir(err, path)
+		}
+		if !info.IsDir() {
+			wg.Add(1)
+			go func(path string) {
+				defer wg.Done()
+				processFile(path, h, &mu, response, provider)
+			}(path)
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+	wg.Wait()
+	return nil
+}
+
+func processFile(path string, h *Handler, mu *sync.Mutex, response *models.RegistryAPIResponse, provider models.Provider) {
+	if meshkitutils.IsZip(path) || meshkitutils.IsTarGz(path) {
+		newTempDir, err := os.MkdirTemp("", "nested-extracted-")
+		if err != nil {
+			incrementCounter(mu, &response.EntityCount.TotalErrCount)
+			h.log.Error(meshkitutils.ErrCreateDir(err, "Error creating nested temp directory"))
+			addUnsuccessfulEntry(path, response, meshkitutils.ErrCreateDir(err, "Error creating nested temp directory"), "")
+			return
+		}
+		defer os.RemoveAll(newTempDir)
+		if err := processUploadedFile(path, newTempDir, h, response, provider, false); err != nil {
+			incrementCounter(mu, &response.EntityCount.TotalErrCount)
+			h.log.Error(err)
+			addUnsuccessfulEntry(path, response, err, "")
+			return
+		}
+		return
+	}
+
+	content, err := os.ReadFile(path)
+	if err != nil {
+		incrementCounter(mu, &response.EntityCount.TotalErrCount)
+		h.log.Error(meshkitutils.ErrReadFile(err, path))
+		addUnsuccessfulEntry(path, response, meshkitutils.ErrReadFile(err, path), "")
+		return
+	}
+	processFileToRegistry(response, content, mu, path, h)
+}
+
+func processFileToRegistry(response *models.RegistryAPIResponse, content []byte, mu *sync.Mutex, path string, h *Handler) {
+	var jsonData interface{}
+	if err := yaml.Unmarshal(content, &jsonData); err == nil {
+		jsonData = utils.ConvertToJSONCompatible(jsonData)
+		convertedContent, err := json.Marshal(jsonData)
+		if err == nil {
+			content = convertedContent
+		}
+	}
+	entityType, err := meshkitutils.FindEntityType(content)
+	if err != nil {
+		incrementCounter(mu, &response.EntityCount.TotalErrCount)
+		addUnsuccessfulEntry(path, response, err, "")
+	}
+	if entityType != "" {
+		path, err := RegisterEntityForImport(content, entityType, h, response, mu)
+		if err != nil {
+			incrementCountersOnErr(mu, entityType, response)
+			h.log.Error(err)
+			addUnsuccessfulEntry(path, response, err, string(entityType))
+		} else {
+			if path != "" {
+				incrementCountersOnSuccess(mu, entityType, &response.EntityCount.CompCount, &response.EntityCount.RelCount, &response.EntityCount.ModelCount)
+				addSuccessfulEntry(content, entityType, response)
+			}
+
+		}
+	}
+}
+func addSuccessfulEntry(content []byte, entityType entity.EntityType, response *models.RegistryAPIResponse) {
```

Comment:
Similar logic for logging and eventing for registration is present, let's try to use that.

Reply:
No this is different.